varianteval/core/variant/sv.py

```diff
@@ -1,40 +1,1200 @@
-class SVCallset:
-    def __init__(self):
-        self.calls = []
-        # interval tree index etc
+"""
+The main data structures for representing variants.
 
-    def add(self, sv):
-        self.calls.append(sv)
+.. image:: sv.png
 
-    def __len__(self):
-        return len(self.calls)
+Remark: phasing is not currently modeled and it is left to future work.
+"""
 
-    def __str__(self):
-        pass
+import copy
+import math
+import pickle
+from typing import Any, Callable
+import pysam
+
+import varianteval.core.constants
+import varianteval.core.vcf as vcf
+import varianteval.core.utils as utils
+import varianteval.core.variant.intervals as intervals
+
+
+# All distinct objects of a given type. The same object might be pointed to by
```

Comment:
why global variables? we should have an SVCallset object to capture this state; usually best to avoid using global variables and in our case we will be working with multiple callsets as well
https://google.github.io/styleguide/pyguide.html#25-global-variables

Reply:
Done. This induced several changes, so the code should be reviewed again.